src/js/hooks/useOnClickOutside.ts

```diff
@@ -0,0 +1,21 @@
+import * as React from 'react';
+
+const useOnClickOutside = (ref, handler) => {
+  const listener = (event) => {
+    if (ref.current && !ref.current.contains(event.target)) {
+      handler(event);
+    }
+  };
+
+  React.useEffect(() => {
+    document.addEventListener('mouseup', listener);
+    document.addEventListener('touchstart', listener);
+
+    return () => {
+      document.removeEventListener('mouseup', listener);
+      document.removeEventListener('touchstart', listener);
+    };
+  }, [ref, handler]);
+};
+
+export { useOnClickOutside };
```

Comment:
Given a ref to a DOM element, this hook will fire a provided callback handler whenever a click is trigger from outside of the element.